src/content/docs/en/reference/integrations-reference.mdx

```diff
@@ -444,12 +444,17 @@ console.log(path) // URL
 
 **When:** Just after the Vite server is created in "dev" mode, but before the `listen()` event is fired. [See Vite's createServer API](https://vitejs.dev/guide/api-javascript.html#createserver) for more.
 
-**Why:** To update Vite server options and middleware.
+**Why:** To update Vite server options and middleware, or enable support for refreshing the content layer.
 
 ```js
-'astro:server:setup'?: (options: { server: vite.ViteDevServer }) => void | Promise<void>;
+'astro:server:setup'?: (options: {
+  server: vite.ViteDevServer;
+  refreshContent: {
+    loaders?: Array<string>;
+    context?: Record<string, any>;
+  };
```

Comment:
Docs below says this is a function. The example has an `await` so I imagine it is also async.
```suggestion
  refreshContent: (refreshOptions: {
    loaders?: Array<string>;
    context?: Record<string, any>;
  }) => Promise<void>;
```

Reply:
[You're right](https://github.com/withastro/astro/blob/40760a8ace13d8be56f3f164f97831c96565c7ee/packages/astro/src/types/public/integrations.ts#L196)!
nit: but we can just use `options` instead of `refreshOptions` (the context is already clear I think and and it shortens a little)

Reply:
Edited, thanks!

Reply:
Thanks! Good spot.